injectproxy/silences.go

```diff
@@ -0,0 +1,158 @@
+package injectproxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+
+	runtimeclient "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
+	"github.com/prometheus/alertmanager/api/v2/client"
+	"github.com/prometheus/alertmanager/api/v2/client/silence"
+	"github.com/prometheus/alertmanager/api/v2/models"
+	"github.com/prometheus/alertmanager/pkg/labels"
+)
+
+func (r *routes) silences(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "GET":
+		r.listSilences(w, req)
+	case "POST":
+		r.postSilence(w, req)
+	}
+}
+
+func (r *routes) listSilences(w http.ResponseWriter, req *http.Request) {
+	var (
+		q        = req.URL.Query()
+		modified = []string{
+			(&labels.Matcher{Type: labels.MatchEqual, Name: r.label, Value: mustLabelValue(req.Context())}).String(),
```

Comment:
nit: could we write this in multiple lines for readability?
```
&labels.Matcher{
  Type: labels.MatchEqual,
  Name: r.label,
  Value: mustLabelValue(req.Context()),
}.String(),
```

Reply:
done